debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/ReadOnlyIncrementalSnapshotIT.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.postgresql;
+
+import static io.debezium.junit.EqualityCheck.LESS_THAN;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.apache.kafka.connect.data.Struct;
+import org.junit.Test;
+
+import io.debezium.config.CommonConnectorConfig;
+import io.debezium.config.Configuration;
+import io.debezium.heartbeat.Heartbeat;
+import io.debezium.junit.SkipWhenDatabaseVersion;
+import io.debezium.junit.logging.LogInterceptor;
+import io.debezium.pipeline.signal.actions.AbstractSnapshotSignal;
+import io.debezium.pipeline.signal.actions.snapshotting.ExecuteSnapshot;
+import io.debezium.pipeline.signal.actions.snapshotting.StopSnapshot;
+import io.debezium.pipeline.signal.channels.KafkaSignalChannel;
+import io.debezium.relational.RelationalDatabaseConnectorConfig;
+import io.debezium.util.Strings;
+
+@SkipWhenDatabaseVersion(check = LESS_THAN, major = 13, minor = 0, reason = "Function pg_current_snapshot() not supported until PostgreSQL 13")
+public class ReadOnlyIncrementalSnapshotIT extends IncrementalSnapshotIT {
```

Comment:
Should we introcude a common `ReadOnlyIncrementalSnapshotIT` class that would be parent for both MySQL and Postgres tests? If net is there an option for utility class used by the read-only test classes?

Reply:
> If net is there an option for utility class used by the read-only test classes?

I was on that path with the idea to have a simple method to send signal and the let the implementation, based on something decide how to send it, but then there was some others changes. 

I ended up with the more pragmatic solution for now and maybe we can think to it on the next `read only` implementation. WDYT?

Reply:
I agree